src/Tests/Microsoft.NET.Build.Tests/GivenWeWantToRequireWindowsForDesktopApps.cs

```diff
@@ -323,6 +323,24 @@ private TestAsset CreateWindowsDesktopReferenceTestAsset(string projectName, str
             return _testAssetsManager.CreateTestProject(testProject);
         }
 
+        [Theory]
+        [InlineData("net5.0", "TargetPlatformIdentifier", "Windows", "Exe")]
+        [InlineData("netcoreapp3.1", "UseWindowsForms", "true", "WinExe")]
+        [InlineData("netcoreapp3.1", "UseWPF", "true", "WinExe")]
+        [InlineData("netcoreapp3.1", "UseWPF", "false", "Exe")]
+        public void It_infers_WinExe_output_type(string targetFramework, string propName, string propValue, string expectedOutputType)
+        {
+            var testProject = new TestProject()
+            {
+                Name = "WinExeOutput",
+                TargetFrameworks = targetFramework,
+                IsSdkProject = true,
+                IsExe = true,
+            };
+            testProject.AdditionalProperties[propName] = propValue;
+            testProject.AdditionalProperties["TargetPlatformVersion"] = "7.0"; // Ensure TargetPlatformVersion is set so we can build with a TargetPlatformIdentifier
```

Comment:
no need to set this now.

Reply:
Sorry, that was the result of a bad merge, should be fixed now.